proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/biometry/impl/BiometricHelper.java

```diff
@@ -0,0 +1,109 @@
+import android.security.keystore.KeyProperties;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.RequiresApi;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.AlgorithmParameterSpec;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+
+import io.getlime.security.powerauth.biometry.BiometricStatus;
+import io.getlime.security.powerauth.exception.PowerAuthErrorCodes;
+import io.getlime.security.powerauth.exception.PowerAuthErrorException;
+
+/**
+ * The {@code BiometricHelper} class provides helper methods for PowerAuth Mobile SDK. The class
+ * is suppose to be used only by the SDK itself.
+ */
+public class BiometricHelper {
+    /**
+     * Translate {@link BiometricStatus} into appropriate {@link PowerAuthErrorException}.
+     *
+     * @param status Status to be translated to the exception.
+     * @return Exception created for the error status. If status is {@link BiometricStatus#OK},
+     *         then {@code IllegalArgumentException} is produced.
+     */
+    public static @NonNull PowerAuthErrorException getExceptionForBiometricStatus(@BiometricStatus int status) {
+        switch (status) {
+            case BiometricStatus.NOT_ENROLLED:
+                return new PowerAuthErrorException(PowerAuthErrorCodes.PA2ErrorCodeBiometryNotAvailable, "Biometric data is not enrolled on the device.");
+            case BiometricStatus.NOT_SUPPORTED:
+                return new PowerAuthErrorException(PowerAuthErrorCodes.PA2ErrorCodeBiometryNotSupported, "Biometry is not supported on the device.");
+            case BiometricStatus.NOT_AVAILABLE:
+                return new PowerAuthErrorException(PowerAuthErrorCodes.PA2ErrorCodeBiometryNotSupported, "Biometry is not available. Try again later.");
+            case BiometricStatus.OK:
+                throw new IllegalArgumentException("Cannot get exception for success status.");
+            default:
+                throw new IllegalArgumentException("Unknown status.");
+        }
+    }
+
+    /**
+     * Create AES/CBC with PKCS7 padding cipher with given secret key.
+     *
+     * @param key Key to be used for encryption and decryption.
+     * @return {@link Cipher} object or null in case of error.
+     */
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    public static @Nullable Cipher createAesCipher(@NonNull SecretKey key) {
+        try {
+            final Cipher cipher = Cipher.getInstance(KeyProperties.KEY_ALGORITHM_AES + "/" + KeyProperties.BLOCK_MODE_CBC + "/" + KeyProperties.ENCRYPTION_PADDING_PKCS7);
+            final byte[] zero_iv = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
```

Comment:
This cannot be changed, because older SDK is already using this setup. Let me explain why.

The zero-iv is OK here, because we're using Cipher basically as a key derivation. We derive a sequence of random bytes into a key, with using biometry-protected key, provided by the system. The derived key is then used to unlock the biometric factor, stored on the device. This happens always in the same direction, so the IV doesn't add any security here.